projects/graph/src/draw.py

```diff
@@ -1,14 +1,42 @@
-"""
-General drawing methods for graphs using Bokeh.
-"""
+import math
 
 from bokeh.io import show, output_file
 from bokeh.plotting import figure
-from bokeh.models import (GraphRenderer, StaticLayoutProvider, Circle, LabelSet,
-                          ColumnDataSource)
+from bokeh.models import GraphRenderer, StaticLayoutProvider, Oval, Rect
+from bokeh.palettes import Spectral8
 
+N = 8
+node_indices = list(range(N))
 
-class BokehGraph:
-    """Class that takes a graph and exposes drawing methods."""
-    def __init__(self):
-        pass  # TODO
+plot = figure(title='Graph Layout Demonstration', x_range=(-1.1, 1.1), y_range=(-1.1, 1.1),
+              tools='', toolbar_location=None)
+
+graph = GraphRenderer()
+
+graph.node_renderer.data_source.add(node_indices, 'index')
+node_colors = ['red', 'blue', 'green', 'yellow',
+               'black', 'pink', 'purple', 'white']
+graph.node_renderer.data_source.add(node_colors, 'color')
+graph.node_renderer.glyph = Rect(height=0.1, width=0.2, fill_color='color')
+
+graph.edge_renderer.data_source.data = dict(
+    start=[0]*N,
+    end=node_indices)
+
+d = dict(
+    start=[0]*N,
+    end=node_indices)
+print(d)
+
+# start of layout code
+circ = [i*2*math.pi/8 for i in node_indices]
+x = [math.cos(i) for i in circ]
+y = [math.sin(i) for i in circ]
+
+graph_layout = dict(zip(node_indices, zip(x, y)))
+graph.layout_provider = StaticLayoutProvider(graph_layout=graph_layout)
+
+plot.renderers.append(graph)
+
```

Comment:
This is a great start! Don't forget to implement a Graph class in graph.py and setup this BokehGraph constructor here to accept Graph as an object. This code you've written here is great and will work as your show method!